In [14]:
import numpy as np
import pandas as pd

In [15]:
data = pd.read_csv("commodity_lastprice.csv")
# data = data[data["date"] >= "2018-01-01"]
data

date  Corn Futures  Crude Oil  Heating Oil   Wheat   Cocoa  \
0     2018-01-02        353.25  60.369999       2.0584  433.50  1936.0   
1     2018-01-03        353.00  61.630001       2.0880  436.00  1907.0   
2     2018-01-04        351.00  62.009998       2.0770  434.00  1905.0   
3     2018-01-05        351.25  61.439999       2.0587  430.75  1895.0   
4     2018-01-07           NaN        NaN          NaN     NaN     NaN   
...          ...           ...        ...          ...     ...     ...   
1394  2022-08-26        665.00  93.060000       4.0076  784.75  2413.0   
1395  2022-08-28           NaN        NaN       4.0183     NaN     NaN   
1396  2022-08-29        683.00  97.010000       3.9099  842.75  2429.0   
1397  2022-08-30        678.50  91.640000       3.8171  820.25  2408.0   
1398  2022-08-31        671.00  89.550000       3.7154  831.50  2427.0   

          Coffee  Natural Gas    Oats  RBOB Gasoline  Rough Rice  \
0     130.199997        3.056  242.50         1.7631      11.535   
1     128.600006        3.008  244.50         1.7974      11.410   
2     129.550003        2.880  245.75         1.8067      11.475   
3     128.449997        2.795  248.50         1.7858      11.565   
4            NaN          NaN     NaN            NaN         NaN   
...          ...          ...     ...            ...         ...   
1394  238.100000        9.296  417.75         2.8513      17.315   
1395         NaN          NaN     NaN         2.8747         NaN   
1396  240.500000        9.353  416.25         2.8776      17.280   
1397  235.200000        9.042  402.00         2.6944      17.430   
1398  239.100006        9.127  404.75         2.6059      17.505   

      Soybean Meal Futures  Soybean Oil Futures  Sugar #11  Newcastle Coal  \
0               318.100006            33.380001  15.330000          102.65   
1               318.799988            33.759998  15.310000          103.80   
2               319.899994            33.740002  15.250000          104.95   
3               321.200012            33.650002  15.080000          105.30   
4                      NaN                  NaN        NaN             NaN   
...                    ...                  ...        ...             ...   
1394            428.500000            70.820000  18.469999          417.40   
1395                   NaN            66.720000        NaN             NaN   
1396            427.100000            66.440000  18.440001          422.75   
1397            424.500000            70.790000  18.100000          414.55   
1398            415.100000            67.480000  17.889999          425.00   

      Barley  
0     1524.0  
1     1489.0  
2     1515.0  
3     1518.5  
4        NaN  
...      ...  
1394  3048.0  
1395     NaN  
1396  3041.0  
1397  3039.5  
1398     NaN  

[1399 rows x 16 columns]

In [3]:
data.columns

Index(['date', 'Corn Futures', 'Crude Oil', 'Heating Oil', 'Wheat', 'Cocoa',
       'Coffee', 'Natural Gas', 'Oats', 'RBOB Gasoline', 'Rough Rice',
       'Soybean Meal Futures', 'Soybean Oil Futures', 'Sugar #11',
       'Newcastle Coal', 'Barley'],
      dtype='object')

In [16]:
data = data.set_index("date")
data.apply(lambda series: series.first_valid_index())

Corn Futures            2018-01-02
Crude Oil               2018-01-02
Heating Oil             2018-01-02
Wheat                   2018-01-02
Cocoa                   2018-01-02
Coffee                  2018-01-02
Natural Gas             2018-01-02
Oats                    2018-01-02
RBOB Gasoline           2018-01-02
Rough Rice              2018-01-02
Soybean Meal Futures    2018-01-02
Soybean Oil Futures     2018-01-02
Sugar #11               2018-01-02
Newcastle Coal          2018-01-02
Barley                  2018-01-02
dtype: object

In [18]:
data.apply(lambda series: series.last_valid_index())

Corn Futures            2022-08-31
Crude Oil               2022-08-31
Heating Oil             2022-08-31
Wheat                   2022-08-31
Cocoa                   2022-08-31
Coffee                  2022-08-31
Natural Gas             2022-08-31
Oats                    2022-08-31
RBOB Gasoline           2022-08-31
Rough Rice              2022-08-31
Soybean Meal Futures    2022-08-31
Soybean Oil Futures     2022-08-31
Sugar #11               2022-08-31
Newcastle Coal          2022-08-31
Barley                  2022-08-30
dtype: object

In [19]:
pd.set_option('display.max_columns', 500)
commodity_table = data.describe()
commodity_table = commodity_table.reindex(sorted(commodity_table.columns), axis=1)

In [20]:
commodity_table

Barley        Cocoa       Coffee  Corn Futures    Crude Oil  \
count  1149.000000  1178.000000  1177.000000   1193.000000  1268.000000   
mean   1897.882071  2434.531409   138.376194    459.152347    62.729338   
std     491.468651   195.719528    48.130366    133.716784    20.477642   
min    1305.500000  1895.000000    86.650002    302.750000   -37.630001   
25%    1518.500000  2314.000000   104.099998    362.750000    52.070000   
50%    1817.500000  2436.000000   118.199997    387.250000    61.355000   
75%    2100.000000  2552.000000   158.699997    557.250000    70.785000   
max    3261.000000  3054.000000   258.350006    818.250000   123.699997   

       Heating Oil  Natural Gas  Newcastle Coal         Oats  RBOB Gasoline  \
count  1263.000000  1268.000000     1191.000000  1176.000000    1262.000000   
mean      2.059178     3.353133      130.711394   371.370323       1.923444   
std       0.760552     1.678360       97.251939   151.565067       0.693580   
min       0.610400     1.482000       48.500000   220.750000       0.411800   
25%       1.698450     2.403000       67.850000   273.187500       1.475800   
50%       1.978600     2.780000       97.150000   300.500000       1.895700   
75%       2.190400     3.735000      141.575000   403.812500       2.160750   
max       5.135400     9.680000      440.000000   807.000000       4.307600   

        Rough Rice  Soybean Meal Futures  Soybean Oil Futures    Sugar #11  \
count  1177.000000           1184.000000          1200.000000  1181.000000   
mean     12.989104            348.653463            41.721317    14.531931   
std       1.940057             54.630010            17.046315     3.039495   
min       9.780000            280.000000            24.990000     9.210000   
25%      11.695000            303.774994            28.590000    12.150000   
50%      12.560000            329.400009            31.700001    13.360000   
75%      13.740000            391.650002            57.947500    17.660000   
max      22.065000            521.900024            90.599998    20.420000   

             Wheat  
count  1191.000000  
mean    615.645147  
std     172.681649  
min     416.500000  
25%     504.250000  
50%     541.000000  
75%     693.000000  
max    1425.250000

In [21]:
commodity_table.to_csv("commodity_table.csv")

<table class="table table-bordered table-hover table-condensed">
<thead><tr><th title="Field #1">Stats</th>
<th title="Field #2">Corn Futures</th>
<th title="Field #3">Crude Oil</th>
<th title="Field #4">Heating Oil</th>
<th title="Field #5">Wheat</th>
<th title="Field #6">Cocoa</th>
<th title="Field #7">Coffee</th>
<th title="Field #8">Natural Gas</th>
<th title="Field #9">Oats</th>
<th title="Field #10">RBOB Gasoline</th>
<th title="Field #11">Rough Rice</th>
<th title="Field #12">Soybean Meal Futures</th>
<th title="Field #13">Soybean Oil Futures</th>
<th title="Field #14">Sugar #11</th>
<th title="Field #15">Newcastle Coal</th>
<th title="Field #16">Barley</th>
</tr></thead>
<tbody><tr>
<td>count</td>
<td align="right">1237</td>
<td align="right">1314</td>
<td align="right">1312</td>
<td align="right">1238</td>
<td align="right">1223</td>
<td align="right">1222</td>
<td align="right">1314</td>
<td align="right">1221</td>
<td align="right">1311</td>
<td align="right">1222</td>
<td align="right">1229</td>
<td align="right">1253</td>
<td align="right">1226</td>
<td align="right">1236</td>
<td align="right">1192</td>
</tr>
<tr>
<td>mean</td>
<td align="right">467.05</td>
<td align="right">63.53</td>
<td align="right">2.12</td>
<td align="right">625.05</td>
<td align="right">2431.06</td>
<td align="right">141.01</td>
<td align="right">3.48</td>
<td align="right">372.06</td>
<td align="right">1.95</td>
<td align="right">13.14</td>
<td align="right">351.08</td>
<td align="right">42.80</td>
<td align="right">14.67</td>
<td align="right">140.95</td>
<td align="right">1939.87</td>
</tr>
<tr>
<td>std</td>
<td align="right">137.63</td>
<td align="right">20.56</td>
<td align="right">0.81</td>
<td align="right">175.98</td>
<td align="right">193.06</td>
<td align="right">49.26</td>
<td align="right">1.78</td>
<td align="right">148.82</td>
<td align="right">0.69</td>
<td align="right">2.05</td>
<td align="right">55.09</td>
<td align="right">17.48</td>
<td align="right">3.06</td>
<td align="right">109.18</td>
<td align="right">529.22</td>
</tr>
<tr>
<td>min</td>
<td align="right">302.75</td>
<td align="right">-37.63</td>
<td align="right">0.61</td>
<td align="right">416.50</td>
<td align="right">1895.00</td>
<td align="right">86.65</td>
<td align="right">1.48</td>
<td align="right">220.75</td>
<td align="right">0.41</td>
<td align="right">9.78</td>
<td align="right">280.00</td>
<td align="right">24.99</td>
<td align="right">9.21</td>
<td align="right">48.50</td>
<td align="right">1305.50</td>
</tr>
<tr>
<td>25%</td>
<td align="right">364.00</td>
<td align="right">52.28</td>
<td align="right">1.74</td>
<td align="right">505.50</td>
<td align="right">2313.50</td>
<td align="right">104.61</td>
<td align="right">2.43</td>
<td align="right">274.25</td>
<td align="right">1.51</td>
<td align="right">11.72</td>
<td align="right">304.40</td>
<td align="right">28.70</td>
<td align="right">12.21</td>
<td align="right">68.18</td>
<td align="right">1529.00</td>
</tr>
<tr>
<td>50%</td>
<td align="right">389.00</td>
<td align="right">61.70</td>
<td align="right">1.99</td>
<td align="right">546.88</td>
<td align="right">2429.00</td>
<td align="right">119.25</td>
<td align="right">2.81</td>
<td align="right">303.25</td>
<td align="right">1.92</td>
<td align="right">12.70</td>
<td align="right">332.60</td>
<td align="right">32.04</td>
<td align="right">13.47</td>
<td align="right">98.35</td>
<td align="right">1829.25</td>
</tr>
<tr>
<td>75%</td>
<td align="right">571.00</td>
<td align="right">72.07</td>
<td align="right">2.24</td>
<td align="right">721.63</td>
<td align="right">2546.00</td>
<td align="right">183.15</td>
<td align="right">3.95</td>
<td align="right">402.50</td>
<td align="right">2.21</td>
<td align="right">14.00</td>
<td align="right">397.90</td>
<td align="right">59.36</td>
<td align="right">17.93</td>
<td align="right">154.29</td>
<td align="right">2133.63</td>
</tr>
<tr>
<td>max</td>
<td align="right">818.25</td>
<td align="right">123.70</td>
<td align="right">5.14</td>
<td align="right">1425.25</td>
<td align="right">3054.00</td>
<td align="right">258.35</td>
<td align="right">9.68</td>
<td align="right">807.00</td>
<td align="right">4.31</td>
<td align="right">22.07</td>
<td align="right">521.90</td>
<td align="right">90.60</td>
<td align="right">20.42</td>
<td align="right">457.80</td>
<td align="right">3261.00</td>
</tr>
</tbody></table>